Homework/app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,89 @@
+class BookingsController < ApplicationController
+    before_action :authenticate_customer!
+    before_action :load_cities,only:[:new,:create]
+    before_filter :load_booking,only:[:show,:destroy]
+    layout 'customers'
+    include BookingsHelper
+
+  def load_booking
+    @booking = Booking.find_by(id:params[:id])
+  end
+
+  def load_cities
+    @city = City.all
+  end
+
+  def index
+    redirect_to customers_path
+  end
+
+  def new
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @booking.customer_id = current_customer.id
+    # begin
+       datetime = DateTime.new(params[:booking]["datetime(1i)"].to_i,
+                               params[:booking]["datetime(2i)"].to_i,
+                               params[:booking]["datetime(3i)"].to_i,
+                               params[:booking]["datetime(4i)"].to_i,
+                               params[:booking]["datetime(5i)"].to_i,0,'+05:30')
+      city = City.find(params[:booking][:city_id])
+      city_bookings_at_same_time = booking_in_time_span(city.bookings,datetime)
+      cleaners_not_available = city_bookings_at_same_time.collect { |booking|
+        booking.cleaner.id }
+      cleaners_in_city = city.cleaners.where.not(id:cleaners_not_available)
+      cleaners_available = cleaners_in_city.collect { |cleaner|
```

Comment:
@neelgohel this seems fine in terms of rails, but check how many queries it will send to database. Should always be done in a way that it fetches data in an optimized query.